src/styles/main.scss

```diff
@@ -1,3 +1,74 @@
+$row-width: 100px;
+$gap: 1px;
+$days: (
+  "mon": 0,
+  "tue": 1,
+  "wed": 2,
+  "thu": 3,
+  "fri": 4,
+  "sat": 5,
+  "sun": 6,
+);
+
+@function calendarWidth($boxWidth, $gap) {
+  @return (($boxWidth * 7) + ($gap * 6));
+}
+
+@mixin boxSize($value) {
+  height: #{$value};
+  width: #{$value};
+}
+
 body {
   margin: 0;
 }
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: 10px;
+  width: calendarWidth($row-width, $gap);
+
+  @each $day, $pos in $days {
+    &--start-day-#{$day} :first-child {
+      margin-left: ($row-width + $gap) * $pos;
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length-#{$i} :nth-child(+ n + #{$i + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    position: relative;
+    box-sizing: border-box;
+    border: 1px solid black;
+    @include boxSize($row-width);
+
+    transition: 500ms;
+
+    @for $day from 1 through 31 {
+      &:nth-child(#{$day}):hover {
+        background-color: purple;
+        color: wheat;
+
+        transform: translateY(-20px);
+      }
+
+      &:nth-child(#{$day})::before {
+        content: "#{$day}";
+        left: 50%;
+        top: 50%;
+
+        font-size: 30px;
+        font-family: Arial, Helvetica, sans-serif;
+        position: absolute;
+
+        transform: translate(-50%, -50%);
```

Comment:
Move these common `::before` styles outside of `@for` (leave only `content`). Check out the compiled css files in `./dist`, these 6 lines will be repeated 31 times, greatly increasing the total css size